### PR TITLE
Fix 1522

### PR DESCRIPTION
**Pull request by trasher**
| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #1522 #1602

### CODE REVIEW

File: inc/supplier.class.php [hunk 1]
```diff
@@ -459,26 +459,32 @@ function showInfocoms() {
...
-            if (($itemtype == 'SoftwareLicense')
-                && ($nb > $_SESSION['glpilist_limit'])) {
-               $linktype  = 'Software';
-               $linkfield = 'softwares_id';
-            }
-            $link_item = new $linktype();
-
             if ($nb > $_SESSION['glpilist_limit']) {
+               if ($itemtype == 'SoftwareLicense') {
+                  // Set $linktype for link to search engine only
```
Comment (orthagh) [inline inc/supplier.class.php]: you keep this part finally ?
Since 9.1, we have a menu entry for SoftwareLicences

File: inc/supplier.class.php [hunk 2]
```diff
@@ -459,26 +459,32 @@ function showInfocoms() {
...
 
+            if ($itemtype == 'Item_DeviceControl') {
+               $query .= "INNER JOIN `glpi_devicecontrols`
+                           ON (`glpi_items_devicecontrols`.`devicecontrols_id`=`glpi_devicecontrols`.`id`)";
+               $linktype = 'DeviceControl';
+               $linkfield = 'devicecontrols_id';
+            }
+
             $linktable = getTableForItemType($linktype);
+            $link_item = new $linktype();
```
Comment (orthagh) [inline inc/supplier.class.php]: getNameField is static, so i think you don't need an instance

File: inc/supplier.class.php (hunk 1 above)
Comment (trasher) [inline inc/supplier.class.php]: Yes, because I'm unsure :) But I can remove it, no problem.

File: inc/supplier.class.php (hunk 2 above)
Comment (trasher) [inline inc/supplier.class.php]: This code was already existing. It was used to call `getFormURLWithID()` which is static also. WIll do the change.

File: inc/supplier.class.php (hunk 1 above)
Comment (orthagh) [inline inc/supplier.class.php]: I think yes

Comment (trasher) [inline inc/supplier.class.php]: Dropped already ;)

**Pull request merged**